apps/oxlint/src/lint.rs

```diff
@@ -166,71 +169,16 @@ impl Runner for LintRunner {
         let paths = walker.paths();
         let number_of_files = paths.len();
 
-        // TODO(perf): benchmark whether or not it is worth it to store the configurations on a
-        // per-file or per-directory basis, to avoid calling `.parent()` on every path.
-        let mut nested_oxlintrc = FxHashMap::<&Path, Oxlintrc>::default();
-        let mut nested_configs = FxHashMap::<PathBuf, ConfigStore>::default();
-
         let handler = GraphicalReportHandler::new();
```

Comment:
NIT: 
The handler is only used for the get_nested_configs. 
Other functions are creating their own handler too.

Reply:
will follow up with this later.

i'm going to make the caller report the diagnostic, not inside the fn